### PR TITLE
Bump api/sdk versions in acceptance tests

### PR DESCRIPTION
**Pull request by ishustava**
Changes proposed in this PR:
- bump consul's api/sdk versions in acceptance tests
- Update go to 1.16 in machine executors and in the test docker image

How I've tested this PR:
- CI

How I expect reviewers to test this PR:
code review

Checklist:
- [ ] Bats tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,7 +14,14 @@ commands:
...
           command: |
-            go get gotest.tools/gotestsum
+            wget https://golang.org/dl/go1.16.5.linux-amd64.tar.gz
+            sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.16.5.linux-amd64.tar.gz
+            rm go1.16.5.linux-amd64.tar.gz
+            echo 'export PATH=$PATH:/usr/local/go/bin' >> $BASH_ENV
+
+            wget https://github.com/gotestyourself/gotestsum/releases/download/v1.6.4/gotestsum_1.6.4_linux_amd64.tar.gz
+            sudo tar -C /usr/local/bin -xzf gotestsum_1.6.4_linux_amd64.tar.gz
+            rm gotestsum_1.6.4_linux_amd64.tar.gz
```
Comment (ishustava) [inline .circleci/config.yml:24]: `go get` will install dependencies too, and that was creating conflicts when we tried to to restore cache (e.g. https://app.circleci.com/pipelines/github/hashicorp/consul-helm/3267/workflows/38b0913e-dde9-4244-aeae-7a4752fa75cc/jobs/14103)

**Review by kschoche — APPROVED** [review]
Thanks for taking care of this!!!

File: .circleci/config.yml (hunk 1 above)
Comment (thisisnotashwin) [reply .circleci/config.yml:24]: this is an excellent find!!

**Review by thisisnotashwin — APPROVED** [review]
This is excellent work here @ishustava !! Thank you so much for taking care of this so quickly!!

**Pull request merged**